### PR TITLE
Revise doc strings and make timeseries API docs available

### PR DESCRIPTION
**Pull request by canerturkmen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1820-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1820/1/index.html

File: timeseries/src/autogluon/timeseries/predictor.py [hunk 1]
```diff
@@ -370,24 +417,28 @@ def save(self) -> None:
...
     def leaderboard(self, data: Optional[TimeSeriesDataFrame] = None) -> pd.DataFrame:
         """Return a leaderboard showing the performance of every trained model, the output is a
         pandas data frame containing the columns,
-        * 'model': The name of the model.
-        * 'score_val': The validation score of the model on the 'eval_metric'.
-            NOTE: Metrics scores always show in higher is better form.
-            This means that metrics such as RMSE or MAPE will have their signs FLIPPED, and values will be negative.
+
+        * ``model``: The name of the model.
+        * ``score_val``: The validation score of the model on the 'eval_metric'.
```
Comment (Innixma) [inline timeseries/src/autogluon/timeseries/predictor.py:432]: lacking description of columns added when data is specified.

**Review by Innixma — APPROVED** [review]
LGTM!

**Comment by szha** [comment]
Job PR-1820-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1820/2/index.html

**Pull request merged**